syntax = "proto3";
import "PacketsId.proto";

package my.proto.package;

message ConnectionRequest
{
	option (package_id) = CONNECTION_REQUEST;
	string 				ip = 1;
	uint32				port = 2;
	string				hostname = 4;
	bool				search = 3;
}

message ConnectionResponse
{
	option (package_id) = CONNECTION_RESPONSE;
	string 				ip = 1;
	uint32				port = 2;
	string				hostname = 3;
}

message ConnectionPing
{
	option (package_id) = CONNECTION_PING;
}

message ConnectionPong
{
	option (package_id) = CONNECTION_PONG;
}

message ConnectionDisconnect
{
	option (package_id) = CONNECTION_DISCONNECT;
}
